sdks/python/apache_beam/utils/retry_test.py

```diff
@@ -18,6 +18,7 @@
 """Unit tests for the retry module."""
 
 import unittest
+import mock
```

Comment:
Is there a need for using `mock`

The tests could be simplified in `DummyClass` takes a list of values to return. and `func` returns or raises them one by one.